sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelFactory.java

```diff
@@ -103,4 +103,9 @@
 
   /** Converts the given string to a {@link Path}. */
   Path toPath(String path);
+
+  /**
+   * Delete files with the given {@code uris} if they exist.
+   */
+  void deleteIfExists(Collection<String> uris) throws IOException;
```

Comment:
(Of course in the new code you're working on, `FileOperations` and `IOChannelFactory` will merge into the same class.)